dockers/docker-snmp/snmpd.conf.j2

```diff
@@ -32,14 +32,25 @@ agentAddress {{ protocol(agentip) }}:[{{ agentip }}]{% if port %}:{{ port }}{% e
 {% if MGMT_INTERFACE is defined %}
 {% for if, ip in MGMT_INTERFACE %}
 {% set agentip = ip.split('/')[0] %}
-agentAddress {{ protocol(agentip) }}:[{{ agentip }}]:161
+{% set protocol = get_protocol(agentip) %}
+{% if protocol == 'udp6' %}
+agentAddress {{ protocol }}:[{{ agentip }}%{{ if }}]:161
```

Comment:
Interface name needs to be used only in link local address scenario. For non link local it is not required. Can you please differentiate?

Reply:
I verified that it works fine for other IPv6 addresses as well, for example, below is the configuration from T0 VS testbed:
```
agentAddress udp:[10.250.0.101]:161
agentAddress udp6:[fec0::ffff:afa:1%eth0]:161
agentAddress udp:[10.1.0.32]:161
agentAddress udp6:[FC00:1::32%Loopback0]:161
```
```
sudo netstat -tulnp | grep 161
tcp        0      0 127.0.0.1:3161          0.0.0.0:*               LISTEN      298761/snmpd        
udp        0      0 10.1.0.32:161           0.0.0.0:*                           298761/snmpd        
udp        0      0 10.250.0.101:161        0.0.0.0:*                           298761/snmpd        
udp6       0      0 fc00:1::32:161          :::*                                298761/snmpd        
udp6       0      0 fec0::ffff:afa:1:161    :::*                                298761/snmpd 

root@vlab-01:/# snmpget -v2c -c public fec0::ffff:afa:1 1.3.6.1.2.1.1.1.0
iso.3.6.1.2.1.1.1.0 = STRING: "SONiC Software Version: SONiC.master.327516-04a6031b2 - HwSku: Force10-S6000 - Distribution: Debian 11.7 - Kernel: 5.10.0-18-2-amd64"
```

Reply:
Conceptually speaking, the scope ID is applicable only for link local address https://learn.microsoft.com/en-us/previous-versions/aa921042(v=msdn.10).
Please refer to RFC https://datatracker.ietf.org/doc/html/rfc6874

The IPv6 Scoped Address Architecture specification [[RFC4007](https://datatracker.ietf.org/doc/html/rfc4007)] offers
   guidance on how the ZoneID affects interface/address selection inside
   the IPv6 stack.  Note that the behaviour of an IPv6 stack, if it is
   passed a non-null zone index for an address other than link-local, is
   undefined.

Please refer section 4 under security.
   To limit this risk, implementations MUST NOT allow use of this format
   except for well-defined usages, such as sending to link-local
   addresses under prefix fe80::/10.  At the time of writing, this is
   the only well-defined usage known.

Reply:
ok, so it will be better to check if ipv6 address starts with Fe80 and use zone id only for that IP?

Reply:
Yes. That should be good